test/ui/views/grades_view_test.dart

```diff
@@ -0,0 +1,192 @@
+// FLUTTER / DART / THIRD-PARTIES
+import 'package:flutter/material.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+// MANAGER
+import 'package:notredame/core/managers/course_repository.dart';
+
+// MODELS
+import 'package:notredame/core/models/course.dart';
+
+// VIEW / WIDGETS
+import 'package:notredame/ui/views/grades_view.dart';
+import 'package:notredame/ui/widgets/grade_button.dart';
+
+// OTHERS
+import '../../helpers.dart';
+import '../../mock/managers/course_repository_mock.dart';
+
+void main() {
+  CourseRepository courseRepository;
+  AppIntl intl;
+
+  final Course courseSummer = Course(
+      acronym: 'GEN101',
+      group: '02',
+      session: 'É2020',
+      programCode: '999',
+      grade: 'C+',
+      numberOfCredits: 3,
+      title: 'Cours générique');
+
+  final Course courseSummer2 = Course(
+      acronym: 'GEN102',
+      group: '02',
+      session: 'É2020',
+      programCode: '999',
+      grade: 'C+',
+      numberOfCredits: 3,
+      title: 'Cours générique');
+
+  final Course courseWinter = Course(
+      acronym: 'GEN101',
+      group: '02',
+      session: 'H2020',
+      programCode: '999',
+      grade: 'C+',
+      numberOfCredits: 3,
+      title: 'Cours générique');
+
+  final Course courseFall = Course(
+      acronym: 'GEN101',
+      group: '02',
+      session: 'A2020',
+      programCode: '999',
+      grade: 'C+',
+      numberOfCredits: 3,
+      title: 'Cours générique');
+
+  final courses = [courseSummer, courseSummer2, courseWinter, courseFall];
+
+  group("GradesView -", () {
+    setUp(() async {
+      intl = await setupAppIntl();
+      setupNavigationServiceMock();
+      courseRepository = setupCourseRepositoryMock();
+    });
+
```

Comment:
Not sure if it's really that necessary, but maybe adding this:
```suggestion
    tearDown(() {
      unregister<CourseRepository>();
    });
```

Reply:
You are completely right!